src/Analyzers/CSharp/CodeFixes/ConvertTypeOfToNameOf/CSharpConvertTypeOfToNameOfCodeFixProvider.cs

```diff
@@ -30,7 +30,7 @@ protected override string GetCodeFixTitle()
             if (node is MemberAccessExpressionSyntax { Expression: TypeOfExpressionSyntax typeOfExpression })
             {
                 var typeSymbol = model.GetSymbolInfo(typeOfExpression.Type, cancellationToken).Symbol.GetSymbolType();
-                return typeSymbol?.GenerateTypeSyntax();
+                return typeSymbol?.GenerateExpressionSyntax();
```

Comment:
nameof argument is an expression. To the end-user this doesn't change anything. This only uses the correct type of syntax nodes for the tests to pass (text representation is still the same in both cases.)